packages/botonic-api/src/data-provider/index.ts

```diff
@@ -4,24 +4,41 @@ import { User } from '@botonic/core/src/models/user'
 import { DynamoDBDataProvider } from './dynamodb-data-provider'
 import { LocalDevDataProvider } from './local-data-provider'
 
+export const dataProviderProtocols = {
+  DYNAMO_DB: 'dynamodb',
+}
+
 export interface DataProvider {
-  getUser(id: string): User | Promise<User> | undefined
+  getUsers(limit?: number, offset?: number): User[] | Promise<User[]>
+  getUser(id: string): User | Promise<User | undefined> | undefined
   saveUser(user: User): User | Promise<User>
   updateUser(user: User): User | Promise<User>
-  getEvent(id: string): any
+  getEvents(
+    limit?: number,
+    offset?: number
+  ): BotonicEvent[] | Promise<BotonicEvent[]>
+  getEvent(
+    id: string
+  ): BotonicEvent | Promise<BotonicEvent | undefined> | undefined
   saveEvent(event: BotonicEvent): BotonicEvent | Promise<BotonicEvent>
-  getUserByWebsocketId(websocketId: string): User | Promise<User>
+  getUserByWebsocketId(
+    websocketId: string
+  ): User | Promise<User | undefined> | undefined
 }
 
-let localDataProvider: LocalDevDataProvider | undefined = undefined
+const localDataProvider: LocalDevDataProvider | undefined = undefined
 // url: dynamodb://my-table.my-region.aws.com
 // url: postgresql://my-database.my-db-provider.com
-export function dataProviderFactory(url: string): DataProvider | undefined {
-  const protocol = url && url.split('://')[0]
+export function dataProviderFactory(url?: string): DataProvider {
   if (!url) {
-    if (!localDataProvider) localDataProvider = new LocalDevDataProvider()
-    return localDataProvider
-  } else if (protocol === 'dynamodb') {
-    return new DynamoDBDataProvider(url)
+    return localDataProvider ?? new LocalDevDataProvider()
+  }
+
+  const protocol = url.split('://')[0]
+  switch (protocol) {
+    case dataProviderProtocols.DYNAMO_DB:
+      return new DynamoDBDataProvider(url)
+    default:
+      return localDataProvider ?? new LocalDevDataProvider()
```

Comment:
Good refactor! Now that we're at it, maybe it makes sense to add a "local database URL" to let developers customize the default local DB. For instance `file://path/to/db.json`. We can also remove the global variable `const localDataProvider` as we needed it to avoid duplicated local DBs, something that it shouldn't be a problem anymore with data persisted to disk.

Reply:
Totally  agree. Done!

I'm also reloading the JSON file before every action made to make sure the DB is always up to date.